common/persistence/src/main/java/app/coronawarn/server/common/persistence/service/DiagnosisKeyService.java

```diff
@@ -113,9 +151,13 @@ public void applyRetentionPolicy(int daysToRetain) {
         .ofInstant(Instant.now(), UTC)
         .minusDays(daysToRetain)
         .toEpochSecond(UTC) / SECONDS_PER_HOUR;
-    int numberOfDeletions = keyRepository.countOlderThanOrEqual(threshold);
+    int numberOfDeletions = keyRepository.countOlderThanOrEqual(threshold, countryCode);
     logger.info("Deleting {} diagnosis key(s) with a submission timestamp older than {} day(s) ago.",
         numberOfDeletions, daysToRetain);
-    keyRepository.deleteOlderThanOrEqual(threshold);
+    keyRepository.deleteOlderThanOrEqual(threshold, countryCode);
+  }
+
+  public void applyRetentionPolicy(int daysToRetain) {
```

Comment:
IT might cause problems if we leave this method as part of the DiagnosisKeyService API, because clients don't know what the behaviour is and which would be the default country. I think it's better if we remove this code path and enforce callers to specify the country. Or we could move the DEFAULT_COUNTRY one level higher and provide it to the service at construction time (as an env variable)